### DISCUSSION TITLE
How to support NBT tags for blocks by loading chunks via AnvilLoader?

### DISCUSSION THREAD

**Discussion by DearFox** [Minestom/Minestom#995] [Help]
I have world regions that I load via AnvilLoader, but alas, the text on the plates and screens for the player's heads are not loaded and are not visible in the game.
How can this be fixed?
Or do I need to prescribe tags for each such block manually?

**Comment by DasLixou** [comment]
you need BlockHandlers. See https://wiki.minestom.net/world/blocks#handlers

so for player heads you do:
```java
public class SkullHandler implements BlockHandler {

    @Override
    public @NotNull NamespaceID getNamespaceId() {
        return NamespaceID.from(Key.key("minecraft:skull"));
    }

    @Override
    public @NotNull Collection<Tag<?>> getBlockEntityTags() {
        return List.of(
                Tag.String("ExtraType"),
                Tag.NBT("SkullOwner")
        );
    }

}
```

and then register it with `MinecraftServer.getBlockManager().registerHandler(Key.key("minecraft:skull"), SkullHandler::new);`

**Reply by DearFox** [reply]
This really works for heads, thanks.

I tried to create the same thing but for spruce_sign (only for one because they are of different types), but for some reason it didn’t work out for me.

```java
import net.kyori.adventure.key.Key;
import net.minestom.server.instance.block.BlockHandler;
import net.minestom.server.tag.Tag;
import net.minestom.server.utils.NamespaceID;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.List;

public class SignHandler implements BlockHandler {

    @Override
    public @NotNull NamespaceID getNamespaceId() {
        return NamespaceID.from(Key.key("minecraft:spruce_sign"));
    }

    @Override
    public @NotNull Collection<Tag<?>> getBlockEntityTags() {
        return List.of(
                Tag.String("ExtraType"),
                Tag.NBT("Color"),
                Tag.NBT("GlowingText"),
                Tag.NBT("Text1"),
                Tag.NBT("Text2"),
                Tag.NBT("Text3"),
                Tag.NBT("Text4")
        );
    }
}
```
I tried to send only 1 NBT tag :

```java
return List.of(
                Tag.String("ExtraType"),
                Tag.NBT("Text1")
        );
```
but it didn't give any results

**Reply by DasLixou** [reply]
When loading an anvil world with a sign, you will see that the namespace ID is only „minecraft:sign“. When you only want to apply it to Spruce Signs, then use `Block.SPRUCE_SIGN` (or so) and only add the handler for it. Also note that sign doesn’t  use Tag.NBT for everything. See https://minecraft.fandom.com/wiki/Sign and then under section Block Data. GlowingText is a Boolean and Color and text1 to text4 are all Tag.String

**Reply by DasLixou** [reply]
So simple: replace namespaceid to `"minecraft:sign"` and make the tags like https://minecraft.fandom.com/wiki/Sign#Block_data
